components/org.wso2.carbon.identity.conditional.auth.functions.http/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/http/util/ClientCredentialAuthConfig.java

```diff
@@ -0,0 +1,392 @@
+                .setSocketTimeout(ConfigProvider.getInstance().getReadTimeout())
+                .setRedirectsEnabled(false)
+                .setRelativeRedirectsAllowed(false)
+                .build();
+
+        try (CloseableHttpClient client = HttpClientBuilder.create().setDefaultRequestConfig(config).build();
+             CloseableHttpResponse response = client.execute(request)) {
+
+            int responseCode = response.getStatusLine().getStatusCode();
+            if (responseCode >= 200 && responseCode < 300) {
+                return processSuccessfulResponse(response);
+            } else {
+                LOG.error("Failed to retrieve access token. Response Code: " + responseCode + ". Session data key: " +
+                        authenticationContext.getContextIdentifier());
+            }
+        } catch (ConnectTimeoutException e) {
+            if (LoggerUtils.isDiagnosticLogsEnabled()) {
+                DiagnosticLog.DiagnosticLogBuilder diagnosticLogBuilder = new
+                        DiagnosticLog.DiagnosticLogBuilder(Constants.LogConstants.ADAPTIVE_AUTH_SERVICE,
+                        Constants.LogConstants.ActionIDs.RECEIVE_TOKEN);
+                diagnosticLogBuilder.inputParam(Constants.LogConstants.InputKeys.TOKEN_ENDPOINT, getTokenEndpoint())
+                        .configParam(Constants.LogConstants.ConfigKeys.SUPPORTED_GRANT_TYPES,
+                                GRANT_TYPE_CLIENT_CREDENTIALS)
+                        .resultMessage("Connection timed out while requesting access token.")
+                        .logDetailLevel(DiagnosticLog.LogDetailLevel.APPLICATION)
+                        .resultStatus(DiagnosticLog.ResultStatus.FAILED);
+                LoggerUtils.triggerDiagnosticLogEvent(diagnosticLogBuilder);
+            }
+            LOG.error("Connection timed out while requesting access token: " + tokenEndpoint, e);
+        } catch (SocketTimeoutException e) {
+            if (LoggerUtils.isDiagnosticLogsEnabled()) {
+                DiagnosticLog.DiagnosticLogBuilder diagnosticLogBuilder = new
+                        DiagnosticLog.DiagnosticLogBuilder(Constants.LogConstants.ADAPTIVE_AUTH_SERVICE,
+                        Constants.LogConstants.ActionIDs.RECEIVE_TOKEN);
+                diagnosticLogBuilder.inputParam(Constants.LogConstants.InputKeys.TOKEN_ENDPOINT, getTokenEndpoint())
+                        .configParam(Constants.LogConstants.ConfigKeys.SUPPORTED_GRANT_TYPES,
+                                GRANT_TYPE_CLIENT_CREDENTIALS)
+                        .resultMessage("Socket timed out while requesting access token.")
+                        .logDetailLevel(DiagnosticLog.LogDetailLevel.APPLICATION)
+                        .resultStatus(DiagnosticLog.ResultStatus.FAILED);
+                LoggerUtils.triggerDiagnosticLogEvent(diagnosticLogBuilder);
+            }
+            LOG.error("Socket timed out while requesting access token: " + tokenEndpoint, e);
```

Comment:
Fix this similar to invoking an external api

Reply:
Resolved with https://github.com/wso2-extensions/identity-conditional-auth-functions/pull/157/commits/01b5e2f5f849d7a4e89d792e28845597544fca39